src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyIFormatProvider.cs

```diff
@@ -309,5 +310,22 @@ static INamedTypeSymbol UnwrapNullableValueTypes(INamedTypeSymbol typeSymbol)
                 return typeSymbol;
             }
         }
+
+        private static bool IsValidParseCall(IInvocationOperation invocationOperation, INamedTypeSymbol? guidType)
+        {
+            var targetMethod = invocationOperation.TargetMethod;
+
+            if (targetMethod.Name != "Parse")
+            {
+                return false;
+            }
+
+            if (guidType != null && targetMethod.ContainingType.Equals(guidType))
```

Comment:
null check is not needed here.

Reply:
I was following the existing pattern on https://github.com/dotnet/roslyn-analyzers/blob/d2ef898a6283858c1224c0114cd87fa27b6cb091/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyIFormatProvider.cs#L290-L291 but of course you are correct.